frontend/src/pages/course/CourseSummaryPage.tsx

```diff
@@ -45,7 +45,8 @@ export function CourseSummaryPage() {
               <h1 className="text-2xl font-bold text-white">
                 Summary View of {courseCode}
               </h1>
-              <CourseGroupTable courseid={courseid||""}/>
+              <CourseGroupTable courseid={courseid || ""} />
+              <DownloadCSVButton courseid={courseid || ""} />
```

Comment:
Great job Calvin, looks great! Only thing I would like to ask for your opinion design wise is whether we should also hide the "Download CSV" button when the table has not loaded yet, such as the example:

![image](https://github.com/user-attachments/assets/edadcfb9-bb20-45dd-baf0-d1604cc22e09)

Feel free to make this modification if you want but your PR is approved!